src/sync-pending-events.ts

```diff
@@ -152,7 +152,8 @@ export class EventTracker {
           await db.updateLastMonitorBlock(r, e);
         }
       }
-      await db.updatelastCheckedBlockNumber(pastEvents.breakpoint);
+      let breakpoint = await getValidBlockNumber(this.source, lastCheckedBlockNumber, pastEvents.breakpoint);
+      await db.updatelastCheckedBlockNumber(breakpoint);
```

Comment:
Don't do this, please return null for getValidBlockNumber and do not do update. DB op is time consuming and we should not do it if not necessary to avoid any further issue.